knotx-core/src/test/java/io/knotx/util/MultiMapConverterTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+import com.googlecode.zohhak.api.Coercion;
+import com.googlecode.zohhak.api.Configure;
+import com.googlecode.zohhak.api.TestWith;
+import com.googlecode.zohhak.api.runners.ZohhakRunner;
+import io.vertx.core.json.JsonObject;
+import io.vertx.rxjava.core.MultiMap;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(ZohhakRunner.class)
+@Configure(separator = ";")
+public class MultiMapConverterTest {
+
+  @Test
+  public void toJsonObject_whenMultiMapEmpty_expectEmptyJsonObject() throws Exception {
+    JsonObject jsonObject = MultiMapConverter.toJsonObject(MultiMap.caseInsensitiveMultiMap());
+    assertThat(
+        jsonObject.size(),
+        equalTo(0));
+  }
+
+  @TestWith({
+      "mapKey1:A,B,C;{\"mapKey1\":[\"A\",\"B\",\"C\"]}",
+      "mapKey1:A,B,C,D;{\"mapKey1\":[\"A\",\"B\",\"C\",\"D\"]}",
+      "mapKey1:A,B,C,D,E;{\"mapKey1\":[\"A\",\"B\",\"C\",\"D\",\"E\"]}",
+      "mapKey1:A|mapKey2:B;{\"mapKey1\":[\"A\"],\"mapKey2\":[\"B\"]}",
+      "mapKey1:A|mapKey2:B|mapKey3:C;{\"mapKey1\":[\"A\"],\"mapKey2\":[\"B\"],\"mapKey3\":[\"C\"]}",
+      "mapKey1:A,B|mapKey2:C,D|mapKey3:E,F;{\"mapKey1\":[\"A\",\"B\"],\"mapKey2\":[\"C\",\"D\"],\"mapKey3\":[\"E\",\"F\"]}",
+      "mapKey1:A,B,C|mapKey2:A,B,C|mapKey3:A,B,C;{\"mapKey1\":[\"A\",\"B\",\"C\"],\"mapKey2\":[\"A\",\"B\",\"C\"],\"mapKey3\":[\"A\",\"B\",\"C\"]}"
+  })
+  public void toJsonObject_whenMultiMapWithMultipleEntries_expectJsonObjectWithJsonArrays(
+      MultiMap multiMap, String expectedJson) throws Exception {
+    JsonObject jsonObject = MultiMapConverter.toJsonObject(multiMap);
+    assertThat(
+        jsonObject.toString(),
+        equalTo(expectedJson));
```

Comment:
Thank you, I didn't know about this lib. Those JSONs are pretty plain, however the syntax of `assertThat - sameJSONAs` is what fits here perfectly.